backend/apps/matching/admin.py

```diff
@@ -23,6 +23,42 @@ def participant_required(function=None, redirect_field_name=REDIRECT_FIELD_NAME,
     return actual_decorator
 
 
+def can_approve_type_in_url(function=None):
+    """
+    Check that the user at hand can approve objects of given p_type
+    """
+
+    def actual_decorator(function):
+        def new_func(request, p_type, *args, **kwargs):
+            if request.user.has_perm(p_type):
```

Comment:
`p_type` here is either wrong or very misleading I think. It should be the name of the permission.

Reply:
Probably something like
```suggestion
            if request.user.has_perm("can_aporive_type_" + p_type.lower()):
```
?